native/src/network.c

```diff
@@ -79,6 +80,17 @@ static apr_status_t sp_socket_cleanup(void *data)
         (*s->net->cleanup)(s->opaque);
     if (s->sock) {
         apr_socket_t *as = s->sock;
+#ifdef APR_UNIX
+        apr_sockaddr_t *sa = NULL;
+        apr_socket_addr_get(&sa, APR_LOCAL, as);
+        if (sa && sa->family == APR_UNIX) {
+            char *sock;
+            apr_getnameinfo(&sock, sa, 0);
+            if (sock) {
+                apr_file_remove(sock, s->pool);
```

Comment:
This isn't an interesting one, the file is deleted *before* the socket is closed?

Reply:
In theory the order shouldn't matter, the socket becomes invisible once deleted.

I've redone the cleanup as there is no way to distinguish between cleaning up the main listening socket and the individual connections. Now the cleanup only runs when the pool created by create is destroyed.